lib/Settings/Admin.php

```diff
@@ -185,12 +185,27 @@ public function getChequebookBalance($urlOptions) {
 		return null;
 	}
 
+	/**
+	 * Buys a new postage batch method by using the client libary handler
+	 * @return \mixed
+	 */
+	public function buyPostageStamp($amount, $depth, $urlOptions) : mixed {
+		$uri = "/stamps/" . $amount . "/" . $depth;
+		$curl = $this->setCurl($uri, $urlOptions);
+		curl_setopt($curl, CURLOPT_POST, 1);
+		$output = curl_exec($curl);
+		$response_data = json_decode($output, true);
+		curl_close($curl);
+
+		return $response_data;
+	}
+
 	/**
 	 * initializes a curl handler
 	 * @return \CurlHandle
 	 */
-	protected function setCurl(string $uri_params, array $urlOptions) : \CurlHandle {
-		$url_endpoint = $urlOptions['ip'] . ":" . $urlOptions['debug_api_port'];
+	protected function setCurl(string $uri_params, array $urlOptions, $debugflag = true) : \CurlHandle {
```

Comment:
`$debugflag` is a bit too vague of a variable name, as it could be related to something other than usage of the debug api. `$useDebugApi` would be a better fit.

Reply:
Corrected